File: src/game/cmds/wraps/PlayerWrap.go
package wraps

import (
	rpc "game/common/rpc/simple"
)

type PlayerWrap struct {
	
	GUID string // 唯一ID
	CreateTime int64 // 创建时间
	rpc.Wrapper
}

func (w *PlayerWrap)Decode(pck *rpc.Packet) rpc.Wrapper {
	
	w.GUID = pck.PopString(); 
	w.CreateTime = pck.PopInt64(); 
	return w
}

func (w *PlayerWrap)Encode(pck *rpc.Packet) {
	
	pck.PutString(w.GUID) 
	pck.PutInt64(w.CreateTime) 
}

File: src/game/tools/rpc/output/wrap/PlayerWrap.ts
class PlayerWrap extends Net.Simple.IWrapper {
	
	public GUID:string; // 唯一ID
	public CreateTime:Long; // 创建时间
	public constructor() {
		super();
	}
	Decode(pck:Net.Simple.Packet):PlayerWrap{
		
		this.GUID = pck.GetString(); 
		this.CreateTime = pck.GetLong(); 
		return this;
	}
    Encode(pck:Net.Simple.Packet){
		
    	pck.PutString(this.GUID); 
    	pck.PutLong(this.CreateTime); 
	}

}